src/color.rs

```diff
@@ -944,12 +953,116 @@ impl Color {
         Color::hsla(h, s_mod, l, self.alpha)
     }
 
+    /// Convert this `Color` to grayscale, which is essentailly desaturating
+    /// it by 100%. For more information on desaturation please see [`desaturate`].
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// // Let's start with a standard HSL color
+    /// let original = Color::hsl(0.0, 0.5, 0.5);
+    /// 
+    /// // Now, when we switch this to grayscale, we will end up with Turtle 'grey'
+    /// assert_eq!(original.grayscale(), "grey".into());
+    /// ```
+    /// 
+    /// A `Color` that has no saturation is known as an '[achromatic]' `Color`, which
+    /// are gray colors ranging from white to black.
+    /// 
+    /// Since this is essentally removing all saturation you can verify that
+    /// the grayscale version of any `Color` is the same `hue` and `lightness` values
+    /// with 0 `saturation`.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// // An arbitrary HSL color
+    /// let original = Color::hsl(200.0, 0.9, 1.0);
+    /// 
+    /// // The grayscale version simply has a saturation of 0
+    /// assert_eq!(original.grayscale(), Color::hsl(200.0, 0.0, 1.0));
+    /// ```
```

Comment:
Awesome! Great work with these docs! I appreciate the use of the "grey" color name and how you succinctly explained everything the user needs to understand about this method. Super!